### ISSUE TITLE
Does not trigger HTML insertion on SPA-like navigation

### ISSUE THREAD

**Issue by jcbhmr** [zvizvi/GitHub-Web-IDE#13]
GitHub uses an SPA-like navigation system which does magic ✨ to avoid triggering a full page reload. This means that the HTML injection that is done once doesn't happen again when a non-navigation-but-still-sorta-navigation event occurs when the user changes "tabs" in the repo.

Video of what I am talking about:
https://user-images.githubusercontent.com/61068799/177011417-7976339c-1971-46e6-b4f3-b3246bfbd4fb.mp4

Listening for some kind of URL/history/navigation-related event and re-triggering the `init()` call _might_ fix it? I don't really know.

Maybe relevant links:
- [Chrome navigation API](https://developer.chrome.com/docs/web-platform/navigation-api/#try-the-navigation-api:~:text=navigation.addEventListener(%27navigatesuccess%27%2C%20event%20%3D%3E%20%7B%0A%20%20loadingIndicator.hidden%20%3D%20true%3B%0A%7D)%3B)
- [Can I use: navigation API](https://caniuse.com/mdn-api_navigation)
- [History popstate](https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event)
- [History pushstate hack](https://stackoverflow.com/a/25673911) (quote: "It also doesn't work in Google Chrome extension content scripts, because it's not allowed to alter the site's JS environment. You can work around that by inserting a <script> with said code, but that's even more overkill.")

There looks to be like 4 more StackOverflow questions basically asking the same thing and all getting basic: "you monkeypatch the history.pushState() method" answers:
- https://stackoverflow.com/a/10419974
- https://stackoverflow.com/a/4585031
- https://stackoverflow.com/a/37492075
- https://stackoverflow.com/a/56760883

**Comment by jcbhmr** [comment]
Possible solution:
```diff
diff --git a/src/js/main.js b/src/js/main.js
index c62ef70..4f44dd0 100644
--- a/src/js/main.js
+++ b/src/js/main.js
@@ -245,3 +245,13 @@ function fixCommitsPageStyle (menuElement) {
 
 window.onload = init;
 init();
+
+// Caniuse: https://caniuse.com/mdn-api_navigation
+// Chrome dev article: https://developer.chrome.com/docs/web-platform/navigation-api/#try-the-navigation-api:~:text=navigation.addEventListener(%27navigatesuccess%27%2C%20event%20%3D%3E%20%7B%0A%20%20loadingIndicator.hidden%20%3D%20true%3B%0A%7D)%3B
+try {
+  navigation.addEventListener('navigatesuccess', event => {
+    init()
+  });
+} catch {
+  // IDK what to do for a fallback
+}
```

**Comment by zvizvi** [comment]
Hi @jcbhmr.
Thank you for opening this issue.
What I did specifically on the GitHub site is listen to the custom event `pjax:end` which is triggered when the page route changes.
This is a good solution only for the GitHub site, as it uses the 'jquery-pjax' library.

https://github.com/zvizvi/GitHub-Web-IDE/blob/861b31ec1ba3b6ba3d44d693d406c92742010a7c/src/js/main.js#L162

**Comment by zvizvi** [comment]
Update:
They seem to have stopped using 'jquery-pjax' recently, so this solution has actually stopped working...
Indeed, other solutions need to be sought 🤔

**Comment by jcbhmr** [comment]
I assume this is a Chrome-only extension? If so, how far back do you want to support? If its Chrome 102+, then the `onnavigatesuccess` is a perfect drop-in substitute for that already existing listener! (I think)

If not, IDK what to do...

Thoughts?

![image](https://user-images.githubusercontent.com/61068799/177208338-96309cf3-4a81-4fb9-a570-e1f93f748ff0.png)

**Comment by jcbhmr** [comment]
Another thing you could use as an indicator is a `MutationObserver` tied to the progress bar that GitHub uses to signal SPA loading... IDK if that is a good idea though.

![image](https://user-images.githubusercontent.com/61068799/177209864-57a791b7-8e1a-4da1-8727-29e7ad734a21.png)
![image](https://user-images.githubusercontent.com/61068799/177209905-dc0a9277-bbad-43af-86f5-364973326f89.png)

**Comment by zvizvi** [comment]
Fixed now in v2.0.3, using the `soft-nav:success` event listener.
Thank you @jcbhmr!
